Libraries/acdriver/Sources/Extensions.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <acdriver/Extensions.h>
+#include <libutil/Options.h>
+
+using acdriver::Extensions;
+
+static std::pair<bool, std::string>
+InsertNextImageType(Extensions::ImageTypeSet &imageTypeSet, std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    ext::optional<std::string> value;
+    auto ret = libutil::Options::Next<std::string>(&value, args, it);
+    if (!ret.first) {
+        return ret;
+    }
+    ext::optional<Extensions::ImageType> type = Extensions::ImageTypeFromFileExtension(*value);
+    if (!type) {
+        return std::make_pair(false, "unknown image type " + *value);
+    }
+    imageTypeSet.insert(*type);
+    return std::make_pair(true, std::string());
+}
+
+ext::optional<std::pair<bool, std::string>> Extensions::ActoolOptions::
+parseArgument(std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    std::string const &arg = **it;
+
+    if (arg == "--allow-extensions") {
+        return libutil::Options::Current<bool>(&_allowExtensions, arg);
+    } else if (arg == "--allow-image-types") {
+        return InsertNextImageType(_allowImageTypes, args, it);
+    } else {
+        return ext::nullopt;
+    }
+}
+
+bool Extensions::ActoolOptions::
+areOptionsValid(acdriver::Result *result) const
+{
+    if (!allowExtensions() && allowImageTypes().size() > 0) {
```

Comment:
Nit: Use !.empty() instead of .size()>0.

```cpp
    if (!allowExtensions() && !allowImageTypes().empty()) { 
```

Reply:
Will do :)